src/daemonizer/posix_fork.cpp

```diff
@@ -21,15 +21,34 @@
 namespace posix {
 
 namespace {
-  void quit(std::string const & message)
+  void quit(const std::string & message)
   {
     LOG_ERROR(message);
     throw std::runtime_error(message);
   }
 }
 
-void fork()
+void fork(const std::string & pidfile)
 {
+  // If a PID file is specified, we open the file here, because
+  // we can't report errors after the fork operation.
+  // When we fork, we close thise file in each of the parent
+  // processes.
+  // Only in the final child process do we write the PID to the
+  // file (and close it).
+  std::ofstream pidfs;
+  if (! pidfile.empty ())
+  {
+    if (access(pidfile.c_str(), F_OK) == 0)
+    {
+      quit("PID file " + pidfile + " already exists. Make sure monerod is not already running.");
+    }
+    pidfs.open(pidfile, std::fstream::out | std::fstream::trunc);
+    if (! pidfs.is_open())
+    {
+      quit("Failed to open specified PID file for writing");
```

Comment:
I don't think so. If the user specifies a pidfile and the daemon can't write one I would consider that a serious problem.

I'm actually surprised you call out this one rather the earlier check that quits if the PID file exists. Especially since its so difficult to ensure that the PID file is removed on termination.